Simplenote/src/main/java/com/automattic/simplenote/NoteEditorFragment.java

```diff
@@ -1234,7 +1240,10 @@ public void run() {
 
     @Override
     public void onSaveObject(Bucket<Note> noteBucket, Note note) {
-        // noop
+        if (mIsPaused) {
+            mNotesBucket.removeListener(this);
+            mNotesBucket.stop();
+        }
```

Comment:
does this get called by `saveNote()`? is it called in a separate thread? why would it be called after pausing?

Reply:
Thanks for your reply. Are we intending to shut down all updates from the server during pause or are we mostly concerned here with sending out our local changes before sleeping?

Reply:
We're mostly concerned about sending out our local changes before pausing/sleeping.  Android could destroy activities and fragments once they have been paused.  If that happens before the local changes are sent, it could result in content loss.

Reply:
Why stop listening in `onPause` and not in `onStop`?

If we switch apps and come back won't we have to start the bucket connection again?
Don't we want updates to be processed in the background?

Reply:
We're not stopping `onPause`.  We're stopping `onDestroy`, which is after `onStop`, if the fragment has been paused already (i.e. `mIsPaused`).

Reply:
> since Android could destroy activities and fragments once they have been paused.

ah, so they don't call `onStop` or `onDestroy` when they do this?

Reply:
It _should_ call `onStop` and `onDestroy`, but `onPause` is more of a guarantee.